units/host_status.go

```diff
@@ -69,16 +69,26 @@ func (j *cloudHostReadyJob) Run(ctx context.Context) {
 	if len(hostsToCheck) == 0 {
 		return
 	}
-	providers := map[string][]host.Host{}
+
+	type providerInfo struct {
```

Comment:
I suggest defining structs outside of functions. Also do these variables need to be upper case?